### DISCUSSION TITLE
STP issue

### DISCUSSION THREAD

**Discussion by prspth** [Kidel/Kathara#291] [Q&A] [answered]
Hello,
I'm trying to run a lab with two switches connected in a loop as follow:
```
switch1[0]="A"
switch1[1]="B"
switch1[image]="kathara/base"

switch2[0]="B"
switch2[1]="A"
switch2[image]="kathara/base"
```
I'm using _brctl_ to enable _stp_ and both switches think they are the roots though I can see the _stp_ traffic on both networks. It looks like both switches are ignoring the _stp_ messages. Could it be because the containers should be connected by _veth_?

**Comment by tcaiazzi** [comment, marked as answer]
Dear @prspth, 

Thanks for reaching out! 

Historically, it is not possible to make `STP` network scenarios, since with the [old network plugin](https://github.com/KatharaFramework/NetworkPlugin/tree/main/bridge) collision domains were implemented through Linux Bridges, that catch STP packets, influencing the algorithm. 

However, with the [new network plugin](https://github.com/KatharaFramework/NetworkPlugin/tree/main/vde), based on `VDE`, it should be possible to use STP. Indeed, the VDE plugin does not affect traffic in any way.

We never tested it as we released the new plugin, but I created an example scenario with the same loop you provide, and it seems to work. 

Here is the example: [stp.zip](https://github.com/KatharaFramework/Kathara/files/15200395/stp.zip)

This is the network topology:

![image](https://github.com/KatharaFramework/Kathara/assets/26346836/31742e63-7d58-4039-9835-f89fbe9f6f77)

This is the STP status on both the switches:
![image](https://github.com/KatharaFramework/Kathara/assets/26346836/af53afd8-ae22-4069-97a3-5be16edf0f32)

On `switch2` we have that `eth0` is in blocking state, so it seems to work.

Please let me know if this helps and if the example works properly. 

Thanks!

**Reply by prspth** [reply]
Dear @tcaiazzi,
Thanks for your quick reply and the scenario. I've checked in my kathara settings the `Docker Network Plugin` is set to `kathara/katharanp_vde` and it is. I ran your scenario and unfortunately switches are ignoring each other:
```
root@switch1:/# brctl showstp br0
 bridge id		8000.0e0cda40065c
 designated root	8000.0e0cda40065c
root@switch2:/# brctl showstp br0
 bridge id		8000.26cd05fabfac
 designated root	8000.26cd05fabfac
```
Here are the output of kathara check:
```
*	Current Manager is: Docker (Kathara)
*	Manager version is: 20.10.22
*	Python version is: 3.11.7 (v3.11.7:fa7a6f2303, Dec  4 2023, 15:22:56) [Clang 13.0.0 (clang-1300.0.29.30)]
*	Kathara version is: 3.7.1
*	Operating System version is: macOS-13.1-arm64-arm-64bit
*	Trying to run container with `kathara/base` image...
Deploying devices...|##################################################################| 1/1
*	Container run successfully.
Deleting devices...|###################################################################| 1/1
```
Is there anything else I should check? Thanks!

**Reply by tcaiazzi** [reply]
Dear @prspth, 

I suspect it is something related to your environment, since on my side I have:
```
root@switch1:/# brctl showstp br0
br0
 bridge id		8000.0eb47daf56a0
 designated root	8000.0eb47daf56a0
...
root@switch2:/# brctl showstp br0
br0
 bridge id		8000.12689d22dc1d
 designated root	8000.0eb47daf56a0
...
```

On which platform are you performing the test?
